node/src/components/linear_chain/state.rs

```diff
@@ -267,6 +264,8 @@ impl LinearChain {
         match self.get_signatures(&block_hash) {
             // Not found in the cache, look in the storage.
             None => vec![Outcome::LoadSignatures(fs)],
+            // We know about the block but we haven't seen any signatures for it yet.
+            Some(signatures) if signatures.proofs.is_empty() => vec![Outcome::LoadSignatures(fs)],
```

Comment:
This is not necessarily required but it would load the signatures from storage into the in-memory cache if the cache was pruned in the meantime.